Extensions/TweenBehavior/JsExtension.js

```diff
@@ -0,0 +1,267 @@
+ *
+ * The file must be named "JsExtension.js", otherwise GDevelop won't load it.
+ * ⚠️ If you make a change and the extension is not loaded, open the developer console
+ * and search for any errors.
+ *
+ * More information on https://github.com/4ian/GDevelop/blob/master/newIDE/README-extensions.md
+ */
+module.exports = {
+  createExtension: function (t, gd) {
+    const extension = new gd.PlatformExtension();
+    extension
+      .setExtensionInformation(
+        "Tween",
+        t("Tween animation"),
+        t("Animate object properties via tweening."),
+        "Matthias Meike",
+        "Open source (MIT License)"
+      )
+      .setExtensionHelpPath("/all-features/tween");
+
+    var tweenBehavior = new gd.BehaviorJsImplementation();
+
+    tweenBehavior.updateProperty = function (
+      behaviorContent,
+      propertyName,
+      newValue
+    ) {
+      if (propertyName === "My first property") {
+        behaviorContent.property1 = newValue;
+        return true;
+      }
+      if (propertyName === "My other property") {
+        behaviorContent.property2 = newValue === "1";
+        return true;
+      }
+
+      return false;
+    };
+
+    tweenBehavior.getProperties = function (behaviorContent) {
+      var behaviorProperties = new gd.MapStringPropertyDescriptor();
+
+      behaviorProperties.set(
+        "My first property",
+        new gd.PropertyDescriptor(behaviorContent.property1)
+      );
+      behaviorProperties.set(
+        "My other property",
+        new gd.PropertyDescriptor(
+          behaviorContent.property2 ? "true" : "false"
+        ).setType("Boolean")
+      );
+
+      return behaviorProperties;
+    };
+
+    tweenBehavior.setRawJSONContent(
+      JSON.stringify({
+        property1: "Initial value 1",
+        property2: true
+      })
+    );
+
+    extension
+      .addBehavior(
+        "TweenBehavior",
+        t("Tween animation"),
+        "TweenBehavior",
+        t("Tween animation"),
+        "",
+        "CppPlatform/Extensions/topdownmovementicon.png",
+        "TweenBehavior",
+        tweenBehavior,
+        new gd.BehaviorsSharedData()
+      )
+      .setIncludeFile("Extensions/TweenBehavior/tweenruntimebehavior.js")
+      .addIncludeFile("Extensions/TweenBehavior/shifty.js");
+    //.addIncludeFile("Extensions/TweenBehavior/LICENSE-MIT.txt");
+
+    // Behavior related
+    extension
+      .addAction(
+        "AddObjectVariableTween",
+        t("Add object variable tween"),
+        t("Add a tween animation for an object variable."),
+        t(
+          "Tween variable _PARAM3_ of _PARAM0_ from _PARAM4_ to _PARAM5_ with easing _PARAM6_ over _PARAM7_ms as _PARAM2_"
+        ),
+        t("Tween animation"),
+        "JsPlatform/Extensions/take_screenshot24.png",
+        "JsPlatform/Extensions/take_screenshot32.png"
+      )
+      .addParameter("objectPtr", t("Object"), "", false)
+      .addParameter("behavior", t("Behavior"), "TweenBehavior")
```

Comment:
As mentioned by Lizard-13, this won't work for multiple objects, because this is a single function that you exposed, you should create instead a function on the behavior:

```
// before:
const behavior = extension
      .addBehavior(

// Add the action to the behavior:

behavior.addAction( "AddObjectVariableTween", ....)
  .addParameter("object", t("Object"), "", false)
  .addParameter("behavior", t("Behavior"), "TweenBehavior", false) //also ALWAYS pass all the parameters to addParameter function, otherwise we risk crashing GDevelop - the C++ code behind this could be called with invalid parameters.
  // ...
  // At the end, enter just the name of the function to be called on gdjs.TweenRuntimeBehavior
  .getCodeExtraInformation()
  .setFunctionName("addObjectVariableTween");
```

`addObjectVariableTween` must finally be declared on the `prototype` of `gdjs.TweenRuntimeBehavior`, which is the JavaScript way of declaring function of a class.

`gdjs.TweenRuntimeBehavior.prototype.addObjectVariableTween = function(...) { ... }`

GDevelop will automatically call your function separately for every object and behavior living on the scene that is picked by the action :)